sda/cmd/api/api.go

```diff
@@ -75,12 +75,14 @@ func setup(config *config.Config) *http.Server {
 	r.GET("/ready", readinessResponse)
 	r.GET("/files", getFiles)
 	// admin endpoints below here
-	r.POST("/file/ingest", isAdmin(), ingestFile)                  // start ingestion of a file
-	r.POST("/file/accession", isAdmin(), setAccession)             // assign accession ID to a file
-	r.POST("/dataset/create", isAdmin(), createDataset)            // maps a set of files to a dataset
-	r.POST("/dataset/release/*dataset", isAdmin(), releaseDataset) // Releases a dataset to be accessible
-	r.GET("/users", isAdmin(), listActiveUsers)                    // Lists all users
-	r.GET("/users/:username/files", isAdmin(), listUserFiles)      // Lists all unmapped files for a user
+	if len(config.API.Admins) > 0 {
```

Comment:
So if there are no admins, all requests to these endpoints would get 404 instead of 403? What's the main reasoning behind this change (curious)?

Reply:
I'm also curious.

Reply:
404 is the standard "not found" response which you get for any query that the app is not designed to handle.

Reply:
yes, I am aware of that, but it's unclear to me why there would be a distinction between the cases
"the user is not an admin but there are other admins" (403) vs 
"the user is not an admin and there are no other admins" (404)

Reply:
I see this as "administrative users are specified" being an alternate way of configuring "administrative API should be enabled" and that way it makes sense to me (even if there was a separate setting for it, I'd still prefer admin APIs to be unreachable if no administrative users are configured).

Reply:
404 == admin endpoints not enabled.
403 == endpoints enabled but user is not an admin.

Reply:
Ok, I think I get your point. Thanks @pontus and @jbygdell !